### PR TITLE
Refactor runtime implementation

### LINKED ISSUE #434: [BUG] make runtime status common
**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

### PR DESCRIPTION
**Pull request by cheyang**
### Ⅱ. Does this pull request fix one issue?

fixes #434 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: api/v1alpha1/common.go [hunk 1]
```diff
@@ -48,3 +48,15 @@ type HCFSStatus struct {
 	// Underlayer HCFS Compatible Version
 	UnderlayerFileSystemVersion string `json:"underlayerFileSystemVersion,omitempty"`
 }
+
+// VersionSpec represents the settings for the Alluxio version that fluid is orchestrating.
+type VersionSpec struct {
+	// Image for Alluxio(e.g. alluxio/alluxio)
```
Comment (frankleaf) [inline api/v1alpha1/common.go:54]: 这里已经下沉到common但是comments还是alluxio是不是得改下

File: api/v1alpha1/status.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+import "github.com/fluid-cloudnative/fluid/pkg/common"
+
+// RuntimeStatus defines the observed state of Runtime
+type RuntimeStatus struct {
+	// config map used to set configurations
+	ValueFileConfigmap string `json:"valueFile"`
+
+	// MasterPhase is the master running phase
+	MasterPhase RuntimePhase `json:"masterPhase"`
```
Comment (TrafalgarZZZ) [inline api/v1alpha1/status.go:26]: `RuntimePhase`和`RuntimeConditionType`是不是也可以移到`status.go`中作为一种通用的属性？

File: api/v1alpha1/common.go (hunk 1 above)
Comment (cheyang) [reply api/v1alpha1/common.go:54]: Done

File: api/v1alpha1/status.go (hunk 2 above)
Comment (cheyang) [reply api/v1alpha1/status.go:26]: Done

**Pull request merged**